visualiser/visualiser.py

```diff
@@ -107,114 +108,147 @@ def make_animation(cls, filename="out.gif", delay=3):
+    def extract_arg_strings(self, *args, **kwargs):
         """
-        Returns function signature arguments as string and function label arguments as string.
-            label_argument string contains only the arguments that are not in ignore_args
-            signature_argument_string contains all the arguments available for the function
+        Returns function signature arguments function label arguments as
+        string.
+        label_args_string contains only the arguments that are not in
+        ignore_args.
+        signature_args_string contains all the arguments available for the
+        function.
         """
-        # If show_argument flag is True(default)
-        # Then argument_string is:
-        # a=1, b=31, c=0
-        signature_argument_string = ', '.join([repr(a) for a in args] +
-                                    [f"{key}={repr(value)}" for key, value in kwargs.items()])
-
-        label_argument_string = ', '.join([repr(a) for a in args] +
-                                                           [f"{key}={repr(value)}"
-                                                            for key, value in kwargs.items()
-                                                            if key not in self.ignore_args])
-
-        # If show_argument flag is False
-        # Then argument_string is:
-        # 1, 31, 0
-        if not self.show_argument_name:
-            signature_argument_string = ', '.join([repr(value) for value in args] +
-                                                 [f"{repr(value)}" for key, value in kwargs.items()])
-
-            label_argument_string = ', '.join([repr(a) for a in args] +
-                                              [f"{repr(value)}"
-                                               for key, value in kwargs.items()
-                                               if key not in self.ignore_args])
-
-        return signature_argument_string, label_argument_string
+
+        def get_kwargs_strings(items, show_key=False, ignore_args=[]):
+            """Returns list of kwargs in string format from given kwargs items
+
+            Args:
+                items (dict_items): kwargs items
+                show_key (bool, optional): show key in string.
+                Defaults to False.
+                ignore_args (list, optional) : list of ignored arguments.
+                Default to [].
+
+            Returns:
+                strings_list: list of kwargs in string format
+            """
+            strings_list = []
+
+            for key, value in items:
+                if key not in ignore_args:
+                    if not show_key:
+                        strings_list.append(f"{repr(value)}")
+                    else:
+                        strings_list.append(f"{key}={repr(value)}")
+
+            return strings_list
+
+        args_string = [repr(a) for a in args]
+        signature_kwargs_string = get_kwargs_strings(
+            kwargs.items(),
```

Comment:
Since we are declaring the method and nesting the function. Can we remove visual noise in passing arguments and parameters because we can access the instance variables and variables from the outer scope in the nested function?
Can you please check that?

Reply:
trying to make it more readable. okay I will update this.